src/RequestHandler.php

```diff
@@ -149,7 +150,7 @@ public function getNextSlave()
             $slave = array_shift($available);
 
             // slave available -> connect
-            if (!$this->tryOccupySlave($slave)) {
+            if (!$this->tryOccupySlave($slave) && !$this->redirectionTries > $this->maxRedirectionTries) {
                 return $this->getNextSlave();
```

Comment:
Also shouldn't return anything here.

Reply:
Actually, the whole function might probably be refactored. Either return when `getNextSlave` is successful or retry or fail finally.